### ISSUE TITLE
HTML Attribute Class not working

### ISSUE THREAD

**Issue by psg2** [vahidhedayati/ajaxdependancyselection#12]
I'm trying to apply bootstrap css to the select, and when I look at the code generated in run time, it doesn't have the attribute Class.
Images with code and result HTML bellow.
![code](https://cloud.githubusercontent.com/assets/14082115/13257480/aa7ed33c-da2d-11e5-82d3-939d09d02596.PNG)
![html](https://cloud.githubusercontent.com/assets/14082115/13257481/aa817632-da2d-11e5-9a23-2983cc3a0ffc.PNG)

**Comment by vahidhedayati** [comment]
looks like I am not passing that element through will update it shortly and update ticket when done. Keep it all set and when I enable it the update to later version will fix the issue.

**Comment by vahidhedayati** [comment]
Please try the 1.1 release. I have also done a test on the test site https://github.com/vahidhedayati/testads http://localhost:8080/myContinent/example and example2 show off the new bigFont class on the select boxes.

**Comment by psg2** [comment]
The css is working properly now, thank you.

But somehow in my project the dependency stopped working and I copied the code from your test site, just adjusting the domain classes names.

![2](https://cloud.githubusercontent.com/assets/14082115/13263333/d31b9400-da47-11e5-8d65-a0abeaed5f2c.PNG)
![3](https://cloud.githubusercontent.com/assets/14082115/13263451/6122324a-da48-11e5-9c07-19a9f882716c.PNG)
![4](https://cloud.githubusercontent.com/assets/14082115/13263452/61222c46-da48-11e5-8442-73483cc4b504.PNG)
![5](https://cloud.githubusercontent.com/assets/14082115/13263453/612380dc-da48-11e5-848e-38d4ffd12784.PNG)

Do you have any ideas what could it be?

The only difference that I did notice was that in your test site, your domain classes that depend on others use the ones that they depend on as attributes and I use belongTo to make the relashionship.

**Comment by vahidhedayati** [comment]
Was it working previously on the older version of the plugin? 

I think it may be down to you defining value = and secondaryValue=

Try removing those two last elements from within the tag.

They preset what to load as select boxes load

**Comment by psg2** [comment]
I tried without value and secondaryValue, still doesn't work. 
About the older version, doesn't work either, but I remember it did worked at first, I probably changed something along the way, but I don't know what. 
There is a version working in Grails 2.x for the same project  in https://redeanimalbeta.herokuapp.com/animal/index

**Comment by vahidhedayati** [comment]
At the moment your primary isn't even listing. Have you opened developer console on browser to see if any java script errors? It may be that jquery isn't loaded 

as for primary object it simply lists domain class and returns result of list. Ensure domain class package name matches...

If your still not having luck upload a sample as in the domain classes for these two and I will make some dummy records 

the belongs to shouldn't make a difference

**Comment by vahidhedayati** [comment]
Also do you want css support for grails 2 version of plugin?

**Comment by psg2** [comment]
The primary is listing.
![1](https://cloud.githubusercontent.com/assets/14082115/13265353/46a3d686-da51-11e5-853f-1d495e33c7e9.PNG)

And I checked the console and the javascript for the select is the same as the one from the grails 2 project.
" jQuery.ajax({type:'POST',data:'id=' + escape(this.value) +'&setId=cidade&filterController=null&filterDisplay=null&filterType=null&filterType2=null&filter=null&filter2=null&prevId=null&bindid=estado.id&collectField=id&searchField=nome&domain2=redeanimalg3.Cidade&controller = autoComplete', url:'/autoComplete/ajaxSelectSecondary',success:function(data,textStatus){estadoUpdate(data);},error:function(XMLHttpRequest,textStatus,errorThrown){}});"

If you have time to add css support for the grails 2 version of the plugin, would be good.

I didn't quite understand your third point, sorry about that. I'm not good in English.

**Comment by psg2** [comment]
I made a new project from scratch just using the domain classes Cidade and Estado and the select worked properly. 
I think the problem in the main project is some conflict between this plugin and the Spring Security plugin.

**Comment by vahidhedayati** [comment]
I have released 0.46 for grails 2 which has the fix added to grails 3 for class= added to any selectPrimary secondary etc. 

My third point I think you mean in regards to your domain classes. If you can't get it to work. (I can't see why not) 

It hasn't worked for me now locally but I now why.. I have enabled AjaxDependacySelection in my application.groovy - this may be your issue too..

grails-app/conf/application.groovy 

```
ads {
    security = "enabled"
    whitelist = [
            [dc:'ajaxdependancyselectexample.MyContinent', collect:'id', search:'continentName'],
            [dc:'ajaxdependancyselectexample.MyCountry', collect:'id', search:'countryName'],
            [dc:'ajaxdependancyselectexample.MyCity', collect:'id', search:'cityName'],
            [dc:'ajaxdependancyselectexample.Departments', collect:'id', search:'name'],
            [dc:'ajaxdependancyselectexample.Employee', collect:'id', search:'name'],
            [dc:'ajaxdependancyselectexample.MyShops', collect:'id', search:'shopName'],
            [dc:'redeanimalg3.Cidade', collect:'id', search:'nome'],
            [dc:'redeanimalg3.Estado', collect:'id', search:'nome']
    ]
}
```

I added yours in and it worked locally.

I will upload the test site witha demo of yours but yes it has had me up trying to figure out why a few released back and it was the pesky security

**Comment by psg2** [comment]
I tried doing what you told and it still didn't work, but then I add these lines 

- grails.plugin.springsecurity.logout.postOnly = false
- grails.plugin.springsecurity.rejectIfNoRule = false
- grails.plugin.springsecurity.fii.rejectPublicInvocations = false
- grails.plugin.databasesession.enabled = false

And is working now.
